doc/designs/edns.md

```diff
@@ -0,0 +1,198 @@
+
+## High-level workflow
+
+Enabling DoT: Administrators enable DoT during initial setup or by updating existing DNS configurations through the CLI.
+
+Certificate Management: FreeIPA generates and assigns TLS certificates to the DNS service. Administrators can also provide custom certificates if needed.
+
+DoT Operations: The DNS components listen for DoT traffic on TCP port 853, providing secure communication to clients.
+
+## How to Use
+
+During the deployment of a FreeIPA server, replica, or client, new options are available to enable DoT support. These options allow administrators to enhance the security of DNS traffic. Here’s how to use the new options:
+
+- Enable DoT: use the `--dns-over-tls` option to enable DoT support during the deployment of clients, servers, or replicas. This option deploys Unbound as a local cache resolver (with /etc/resolv.conf listening on 127.0.0.1) and configures BIND on the server to receive DoT requests. On the client side, only Unbound will be deployed (with /etc/resolv.conf listening on 127.0.0.53). Replica deployment configuration will depend on whether the Integrated DNS service is deployed. If it is, the same server configuration will apply. If the Integrated DNS server is not deployed, the client configuration will apply.
+
+- Specify an Upstream DNS Server with DoT enabled: use the `--dot-forwarder` option to specify the upstream DNS server that supports DoT. The format must be 1.2.3.4#dns.server.test. You still need to specify at least one of `--forwarder`, `--auto-forwarders`, or `--no-forwarders` options for the non-encrypted communication as well as discovery process.
+
+- DoT Certificates. If you prefer to use certificates for DoT in Bind/Unbound, use the `--dns-over-tls-key` and `--dns-over-tls-cert` options. If these options are not specified, the IPA CA will be used to request a new certificate.
+
+- DNS Policy: use `--dns-policy` to define the DNS security policy for FreeIPA deployments. It accepts two values: `relaxed` and `enforced`. When set to `relaxed`, the system will attempt to use DoT but will fall back to unencrypted DNS if DoT is unavailable. This mode ensures compatibility with environments where encrypted DNS is not fully supported. When set to `enforced`, the system strictly requires DoT, and any DNS resolution that does not support encryption will be rejected, including discovery from clients.
```

Comment:
Kinda late now but "strict" might be a better term than enforced.

Reply:
Uhmm... I think I was using 'strict,' and we switched to 'enforced'... this would require changes in code.

Reply:
It's fine then, leave it as-is.